docs/proposals/job_controller.md

```diff
@@ -0,0 +1,226 @@
+  The third part is the duration between successive runs of the scheduled pod(s).
+  * "suspendOnFailure": dictated if job schedule should be suspended in case of failure of the last completed job run.
+  * "skipOutdated": dictates if pending scheduled job runs should be discarded in favour of the latest one. This covers the case of a job having suspended its execution after a failed run, with several schedule points reached in the meantime (or the case of a long lasting run with concurrent execution disable). Instead of stacking pending executions, only the latest one is kept (and intermediate pending runs discarded).
+
+**Implementation details**
+
+Job objects are stored in a dedicated registry directory key (separated from the directory key used for replication controllers).
+
+Similarly to replication controllers, a dedicated manager takes care of all jobs existing in the system, performing the following tasks:
+
+* Executes a loop on the full list of defined jobs, calling on each a defined synchronization call-back, each execution loop being spaced by a configuration defined time interval (with default value of 5 seconds).
+* Performs a (recursive) watch of the registry directory key containing jobs definition. For each job newly defined or updated in the system, the same synchronization routine is invoked.
+
+This job manager provides a way for each job defined with schedule information to setup a dedicated timer for next schedule time. This allows a better accuracy of start execution time for pods launched by jobs and remove dependency on the interval value set for the synchronization loop.
+
+The maximal restart count and maximal execution time for pods containers started by a job will be enforced at kubelet level, using dedicated resource limits added by the job in pod's containers definition. Two new resource types ("execution time" and "execution count") are introduced, that are used to add in the container's resources limits the figures defined in job specification and taken into account by the kubelet module (container is killed if the execution time is reached and restarted in case of failure exit up to this maximal count).
+
+In order to precisely track job pods, 2 specific labels are pushed in created pod metadata: 
+* the job name ("jobName")
+* A tag ("jobTag", either a generated UID, or simply the schedule date of this job execution) allowing to group all pods started started for a given job execution.
+
+Along with labels, specific environment variables are also dynamically added in created pods definition: the number of pods started for this job execution and for each one its index number in the started pods list.
+
+Job has the responsibility to advertise job run completion status (success or failure) through events, to perform the necessary clean-up in pods registry, to provide a detailed status of active job runs and the completion status of previous completed runs up to a given amount of history).
```

Comment:
Events are not guaranteed to reach its destination. I agree about the responsibility to advertise, but its the role of the manager (JobController here) to clean-up pods registry, provide detailed status of all jobs at any time.